src/classes/HH_HouseholdNaming_BATCH.cls

```diff
@@ -85,7 +85,7 @@ public class HH_HouseholdNaming_BATCH implements Database.Batchable<SObject> {
     * @return void  
     ********************************************************************************************************/
     public void execute(Database.BatchableContext bc, Sobject[] result) {
-        HH_ProcessControl.inFutureContext = true;     
+        HH_ProcessControl.setRecursionFlag(true);
```

Comment:
doesn't the execute method now need to clear the recursion flag at the end, rather than assuming it goes out of context?